### ISSUE TITLE
Conflicting flag between r5py and pytest

### ISSUE THREAD

**Issue by SergioRec** [datasciencecampus/transport-network-performance#196]
**Describe the bug**
When running pytest through GitHub Actions using `pytest -m runinteg --runinteg --deselect tests/osm/`, run fails with the following traceback:

```
________ ERROR collecting tests/analyse_network/test_analyse_network.py ________
../../../hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/r5py/util/memory_footprint.py:157: in _get_max_memory
    value, unit = _parse_value_and_unit(max_memory)
../../../hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/r5py/util/memory_footprint.py:93: in _parse_value_and_unit
    value = float(matches["value"])
E   TypeError: 'NoneType' object is not subscriptable

[...]

ERROR tests/analyse_network/test_analyse_network.py - ValueError: Could not interpret `--max-memory` ('runinteg').
```

Traceback points to a module in `r5py.util` called [`memory_footprint.py`](https://github.com/r5py/r5py/blob/main/src/r5py/util/memory_footprint.py).

This module defines a command line argument with flags `-m` or `--max-memory`:
```
config.argparser.add(
    "-m",
    "--max-memory",
    help="""
        Memory limit for the JVM running R5.

        Use % as a suffix to specify a share of total RAM;
        K, M, G, T to specify KiB, MiB, GiB, or TiB, respectively.
        Values without suffix are interpreted as bytes.
    """,
    default="80%",
)
```

The abbreviated `-m` flag conflicts with pytest `-m` flag, and r5py is trying to pass the argument `runinteg` into `_get_max_memory`, causing the error.

**Environment**
 - OS: macOS Ventura
 - Python version: 3.9
 - Java version: openjdk 11.0.2 2019-01-15
 - Python module versions (if different from `requirements.txt`): r5py==0.1

**Steps to reproduce**
Error happening in branch [`analyse_network`](https://github.com/datasciencecampus/transport-network-performance/tree/analyse_network) on push.

Error happens during GitHub Actions run.
Pytest runs fine locally if using `pytest tests/analyse_network/`, but give same error as above if running `pytest -m runinteg tests/analyse_network/`

[Full traceback](https://github.com/datasciencecampus/transport-network-performance/actions/runs/6709433402/job/18232468100)

**Comment by SergioRec** [comment]
Tried adding `--max-memory` argument to workflow config file, does not recognise argument.
Tried adding `sys.argv.append(["--max-memory", "99%"])` to test file, same error as described in issue.

**Comment by SergioRec** [comment]
`pytest --runinteg` does not work to select only flagged tests. `pytest -m runinteg` is necessary.

`--deselect` doesn't prevent test collection, so `pytest -m runinteg --deselect tests/analyse_network` fails as:
- wrong version of java is installed in `all-os-tests.yml`
- `Could not interpret '--max-memory' ('runinteg')` in `python-package.yml'`

**Comment by SergioRec** [comment]
Runners have been fixed as part of #198.

However, issue remains that tests using r5py can't be run using `-m` flag in command line.
